### DISCUSSION TITLE
Nodemailer on vercel, next.js

### DISCUSSION THREAD

**Discussion by AyanamiFRei** [vercel/community#1231] [Help]
### Summary

Hello, please help. I'm trying to implement sending mail through nodemailer with serverless functions, everything works and is sent to localhost, but it doesn't work on vercel without any errors in the logs

### Additional information

[POST] /api/contact
11:42:50:55
{
  name: 'eee',
  email: 'ze.kirilov@gmail.com',
  message: 1,
  email_teach: 'k1rilovz@yandex.ru'
}
fetched

**Comment by swarnava** [comment]
Can you confirm if you function is returning response [within limit](https://vercel.com/docs/concepts/limits/overview#general-limits)?  Here is an example using promise:

```js
{

    const transporter = nodemailer.createTransport({
        host: process.env.MAIL_HOST,
        port: 587,
        secure: false,
        auth: {
            user: process.env.MAIL_USERNAME,
            pass: process.env.MAIL_PASSWORD
        }
    });

    let info = await transporter.sendMail({
        from: `"My Name 👻" <${process.env.MAIL_USERNAME}>`,
        to: "foo@example.com, bar@example.com",
        subject: "Hello ✔",
        text: "Hello world?",
        html: "<b>Hello world?</b>",
    });

    if (info.messageId) {
        return {
            statusCode: 200,
            body: nodemailer.getTestMessageUrl(info)
        }
    }
  
    return {
        statusCode: 400,
        body: "Oops"
    }
```

**Reply by AyanamiFRei** [reply]
I am don't sure, i think all within limits. Here is my code and api file:
`const handleSubmit = (e) => {
    e.preventDefault()
    console.log('Sending')

    let data = {
        name,
        email,
        message,
        email_teach
    }

    fetch('/api/contact', {
      method: 'POST',
      headers: {
        'Accept': 'application/json, text/plain, */*',
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(data)
    }).then((res) => {
        console.log('Response received')
        if (res.status === 200) {
            console.log('Response succeeded!')
            setSubmitted(true) 
            setName('1')
            setEmail('2')
            setMessage('3')
            setEmail_teach('4')
        }
    })
  }`
  `
import nodemailer from "nodemailer"
export default async function (req, res) {
    require('dotenv').config()

    let transporter = nodemailer.createTransport({
      host: "smtp.mail.ru",
      port: 465,
      secure: true, // true for 465, false for other ports
      /*auth: {
        user: 'k1rilovz@yandex.ru',
        pass: '---',
      },*/
      auth: {
        user: 'quadratic.eq@mail.ru',
        pass: '---',
      },
    });
    //console.log(transporter)
    const mailData = {
        from: 'quadratic.eq@mail.ru',
        to: `${req.body.email}, ${req.body.email_teach}`,
        subject: `Message From quadratic.eq`,
        text: "Оценка ученика " + req.body.name + " — " + req.body.message,
        html: `<div>Оценка ученика ${req.body.name}  —  ${req.body.message}</div>`
    }
    
    await transporter.sendMail(mailData, function (err, info) {
        if(err)
          console.log(err)
        else
          console.log(info);
    })
  
    console.log(req.body)
    res.send('success')
    console.log('fetched')
  }
`

**Reply by swarnava** [reply]
Do you have a [minimal](https://stackoverflow.com/help/minimal-reproducible-example) repository that we can look into?

**Reply by AyanamiFRei** [reply]
so small, as possible: https://github.com/Sahar8-coder/tex_ex_test
